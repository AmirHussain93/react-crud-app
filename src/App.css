.add-form {
  width: 50%;
  margin: 0 auto;
  padding-top: 60px;
}

.form-group-child {
  width: 65%;
  margin: 0 auto;
}

.addform-inputs {
  width: 100%;
  height: 35px;
  margin-bottom: 10px;
  font-size: 16px;
  padding: 0px 5px !important;
}

.add-form-btn {
  background-color: blue;
  padding: 5px;
  font-size: 16px;
  color: #ffffff;
  border: 1px solid blue;
  border-radius: 5px;
  height: 39px;
  float: right;
  margin-top: 20px !important;
}

.add-form > .bx--form > .bx--fieldset > .bx--label {
  text-align: center;
  font-size: 16px;
  margin-bottom: 25px;
}

.employee-list-table {
  width: 80%;
  float: right;
  margin-top: 6%;
  margin-right: 1%;
}

.add-employee-btn {
  margin-right: 20px !important;
  margin-bottom: 10px !important;
  padding-left: 12px !important;
  padding-right: 12px !important;
}

.sidebar {
  background: #f4f4f4 !important;
  width: 18% !important;
  float: left;
}

.edit-employee {
    padding: 5px;
    margin-right: 10px !important;
    margin-top: 5px !important;
    margin-bottom: 5px !important;
    min-height: 2rem !important;
    padding-left: 20px !important;
    padding-right: 20px !important;
    padding-top: 8px !important;
    padding-bottom: 8px !important;
    width: 65px;
}

.delete-employee {
    background: red !important;
    padding: 5px;
    margin-right: 10px !important;
    border: 1px solid red !important;
    margin-top: 5px !important;
    margin-bottom: 5px !important;
    width: 50px;
    min-height: 2rem !important;
    padding-left: 13px !important;
    padding-right: 13px !important;
    padding-top: 8px !important;
    padding-bottom: 8px !important;
    width: 65px;
}

.imp-field {
  color: red;
}
