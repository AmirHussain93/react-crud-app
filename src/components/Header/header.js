import React from 'react';
import { Button } from 'react-bootstrap';

import './header.scss';


function Header() {
    return (
        <div className="header-section">
            <div className="inner-box">
                <Button className="header-items">Employees</Button>
                <Button className="header-items">Add Employee</Button>
            </div>
        </div>
    )
}

export default Header