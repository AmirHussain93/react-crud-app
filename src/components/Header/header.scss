@import "~bootstrap/scss/bootstrap";

.header-section {
    width: 100%;
    background-color: #000;

    .inner-box {
        padding: 5px;;
    }

    .header-items {
        margin-left: 10px;
    }
}