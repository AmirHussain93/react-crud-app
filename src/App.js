import React from 'react';
import {
  BrowserRouter as Router,
  Switch,
  Route,
  Redirect,
} from "react-router-dom";

import './App.css';
import Header from './components/Header/header';
import Employees from './components/Employees/employees';

function App() {
  return (
    <div className="App">
      <Router>
        <Header />
        <Switch>
          <Route exact path='/'>
            <Redirect to='/employees' />
          </Route>
          <Route exact path='/employees' component={Employees} />
        </Switch>
      </Router>
    </div>
  )
}

export default App;
