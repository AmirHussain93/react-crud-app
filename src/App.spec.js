import React from 'react';
import { mount, shallow } from 'enzyme';
import App from './App';
import Header from './components/Header/header';

describe('App', () => {
  let wrapper;

  beforeEach(() => wrapper = shallow(<App />))

  it('should render correctly', () => expect(wrapper).toMatchSnapshot());

  it('should render a <div />', () => {
    expect(wrapper.find('div').length).toEqual(1);
  });

  it('should render .App', () => {
    expect(wrapper.find('.App').exists());
  })

  it('should render header component', () => {
    expect(wrapper.containsMatchingElement(<Header />)).toEqual(true);
  })
})
