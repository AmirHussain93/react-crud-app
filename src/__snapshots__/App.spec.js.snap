// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly 1`] = `
<div
  className="App"
>
  <BrowserRouter>
    <Header />
    <Switch>
      <Route
        exact={true}
        path="/"
      >
        <Redirect
          to="/employees"
        />
      </Route>
      <Route
        component={[Function]}
        exact={true}
        path="/employees"
      />
    </Switch>
  </BrowserRouter>
</div>
`;
